src/features/world/scenes/BaseScene.ts

```diff
@@ -403,38 +419,42 @@ export abstract class BaseScene extends Phaser.Scene {
   }
 
   moveOtherPlayers() {
-    for (const sessionId in this.playerEntities) {
-      const room = this.roomService.state.context.rooms[this.roomId];
+    const room = this.roomService.state.context.rooms[this.roomId];
+    if (!room) return;
 
-      if (!room) continue;
+    // Destroy any orphaned players
+    Object.keys(this.playerEntities).forEach((sessionId) => {
+      if (!room.state.players.get(sessionId)) {
+        this.destroyPlayer(sessionId);
+      }
+    });
 
-      if (sessionId === room.sessionId) continue;
+    // Render current players
+    room?.state.players.forEach((player, sessionId) => {
+      if (sessionId === room.sessionId) return;
 
-      console.log(sessionId);
       const entity = this.playerEntities[sessionId];
 
-      const position = room.state.players.get(sessionId);
+      // Skip if the player hasn't been set up yet
+      if (!entity.active) return;
 
-      if (!position) {
-        return;
-      }
```

Comment:
This was a bug causing to the render loop if any position was undefined. This occurs when we have stale Bumpkin Containers in the list. I have updated this loop to rely on the game state rather than local state.